Migrations file handling

zUniQueX opened the pull request:
###### Problem:
Liquibase 4.x.x changed its file finding and handling. Therefore migration files in the `dropwizard-migrations` module were not found.

###### Solution:
Add root paths to the `FileSystemResourceAccessor`. Liquibase suggests adding the path `'/'` as a fix when migrating to 4.x.x. This seems to work on Linux, but not on Windows. So this PR adds the operating system specific root path (`C:\` for Windows and `/` otherwise). Additionally, the location of the code source is added (acquired through the `CloseableLiquibase` class).

This allows finding files with an absolute path and files located under `src/main/resources`, that are packaged into the jar as resources. Previously, it was possible to provide paths relative to the root folder of the project, which now isn't possible any more. However, the sources are unlikely to be deployed, so this should be enough in my opinion, when adding this behavior change to the migration guide. This also includes changes to be made to `.xml` files including other ones, as the include mechanism uses the same resolving strategy.

###### Result:
Users can specify custom files to the `dropwizard-migrations` module instead of using the default `migrations.xml`.

Resolves #5150

zUniQueX committed:
Re-enable test for filesystem migrations file

zUniQueX committed:
Add root paths to fix a bug where migration files were not found

zUniQueX committed:
Merge branch 'release/2.1.x' into migrations-file-handling

joschi commented on dropwizard-migrations/src/main/java/io/dropwizard/migrations/CloseableLiquibaseWithFileSystemMigrationsFile.java:
```diff
@@ -17,9 +21,25 @@ public class CloseableLiquibaseWithFileSystemMigrationsFile extends CloseableLiq
+
+    private static List<File> getRootPaths() {
+        List<File> rootPaths = new ArrayList<>();
+        boolean isWindows = System.getProperty("os.name", "").toLowerCase().contains("win");
+        if (isWindows) {
+            rootPaths.add(new File("C:\\"));
+        } else {
+            rootPaths.add(new File("/"));
+        }
```
Should we let Java NIO do the hard work and do something like this?
```suggestion
        rootPaths.add(Paths.get("/").toFile());
```

https://docs.oracle.com/javase/tutorial/essential/io/pathOps.html#create

zUniQueX replied:
Thanks for the suggestion. But in the linked docs I can't read that using `/` would work on Windows too.

After looking deeper into the docs about that, I would change the code to use [`FileSystems.getDefault().getRootDirectories()`](https://docs.oracle.com/javase/7/docs/api/java/nio/file/FileSystem.html#getRootDirectories()). This way we would avoid to deal with OS specific paths and would additionally cover situations with multiple drives.

joschi replied:
Yes, let's use `getRootDirectories()`. 👍

joschi committed:
Use FileSystem#getRootDirectories() in CloseableLiquibaseWithFileSystemMigrationsFile

https://docs.oracle.com/javase/7/docs/api/java/nio/file/FileSystem.html#getRootDirectories()

joschi committed:
Merge branch 'release/2.1.x' into migrations-file-handling

joschi reviewed (approved):
Thanks!

zUniQueX commented:
Looks good! :+1: Should we keep the issue open to have a reminder for the migration guide?
